sql/hive/src/test/scala/org/apache/spark/sql/sources/hadoopFsRelationSuites.scala

```diff
@@ -100,6 +104,87 @@ abstract class HadoopFsRelationTest extends QueryTest with SQLTestUtils {
     }
   }
 
+  test("test all data types") {
+    withTempDir { file =>
+      file.delete()
```

Comment:
You can use `withTempPath` here.  It provides a temporary path without creating the directory, so that you don't need the `delete()` call.